include/circt/Dialect/RTL/RTLCombinatorial.td

```diff
@@ -339,3 +339,53 @@ def MuxOp : RTLOp<"mux",
     }]>
   ];
 }
+
+def BitArrayType : Type<CPred<[{
+  $_self.isa<ArrayType>() &&
+  $_self.cast<ArrayType>().getElementType().isSignlessInteger(1)
+}]>, "RTL array of bits">;
+
+def KnownBitWidthType : Type<CPred<[{getBitWidth($_self) != -1}]>,
+  "Type wherein the bitwidth in hardware is known">;
+
+def CastToBitsOp: RTLOp<"cast.tobits", [NoSideEffect]> {
```

Comment:
I would name this `rtl.bitcast`.  It is a term of art across LLVM IR and has even been adopted by the [C++ community](https://en.cppreference.com/w/cpp/numeric/bit_cast).